spec/unit/type/logical_volume_spec.rb

```diff
@@ -0,0 +1,337 @@
+        :volume_group  => 'elmo',
+        :size          => '10M',
+        :stripes      => '7+'
+      )
+    }.to_not raise_error
+  end
+
+  it 'invalid stripesize raises error' do
+    expect {
+      resource = Puppet::Type.type(:logical_volume).new(
+        :name         => 'bert',
+        :ensure        => :present,
+        :volume_group  => 'ernie',
+        :size          => '10M',
+        :stripesize    => 12345
+      )
+    }.to raise_error(Puppet::ResourceError,
+                     'Parameter stripesize failed on Logical_volume[bert]: 12345 is not a valid stripesize')
+  end
+
+  it 'valid stripe count does not raise error' do
+    expect {
+      resource = Puppet::Type.type(:logical_volume).new(
+        :name         => 'big bird',
+        :ensure        => :present,
+        :volume_group  => 'elmo',
+        :size          => '10M',
+        :stripesize    => '7+'
+      )
+    }.to_not raise_error
+  end
+
+  it 'invalid readahead raises error' do
+    expect {
+      resource = Puppet::Type.type(:logical_volume).new(
+        :name         => 'bert',
+        :ensure        => :present,
+        :volume_group  => 'ernie',
+        :size          => '10M',
+        :readahead    => 'cookie monster'
+      )
+    }.to raise_error(Puppet::ResourceError,
+                     'Parameter readahead failed on Logical_volume[bert]: cookie monster is not a valid readahead count')
+  end
+
+  it 'valid readahead does not raise error' do
+    expect {
+      resource = Puppet::Type.type(:logical_volume).new(
+        :name         => 'bert',
+        :ensure        => :present,
+        :volume_group  => 'ernie',
+        :size          => '10M',
+        :readahead    => '7Auto'
+      )
+    }.to_not raise_error
+  end
+
+  it 'size is minsize raises error if not a boolean' do
+    expect {
+      resource = Puppet::Type.type(:logical_volume).new(
+        :name              => 'kuzco',
+        :ensure            => :present,
+        :volume_group      => 'kronk',
+        :size              => '10M',
+        :size_is_minsize  => 'pacha',
+      )
+    }.to raise_error(Puppet::ResourceError,
+                     'Parameter size_is_minsize failed on Logical_volume[kuzco]: size_is_minsize must either be true or false')
+  end
+
+  it 'size is minsize does not raise error if boolean' do
+    expect {
+      resource = Puppet::Type.type(:logical_volume).new(
+        :name              => 'kuzco',
+        :ensure            => :present,
+        :volume_group      => 'kronk',
+        :size              => '10M',
+        :size_is_minsize  => 'true'
+      )
+    }.to_not raise_error
+  end
+
+  it 'mirror number outside of 0-4 range throws error' do
+    expect {
+      resource = Puppet::Type.type(:logical_volume).new(
+        :name              => 'scooby doo',
+        :ensure            => :present,
+        :volume_group      => 'shaggy',
+        :size              => '10M',
+        :mirror            => -1
```

Comment:
Integers from puppet versions older than 4.0.0 can't actually pass integers; they only pass strings. Maybe check that strings work? (iirc anyway)